src/Features/Core/Portable/ExtractMethod/AbstractExtractMethodService.cs

```diff
@@ -36,6 +38,15 @@ internal abstract class AbstractExtractMethodService<TValidator, TExtractor, TRe
                 return new FailedExtractMethodResult(selectionResult.Status);
             }
 
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            if (localFunction && syntaxFacts.ContainsGlobalStatement(root))
```

Comment:
this is weird. so if i'm extracting to a local function from within a method in a class in a file that hsa global statements, i'm just blocked?

Reply:
At the moment yes. I agree it's weird experience :-/

`MethodExtractor.CodeGenerator.GenerateAsync` (line  96-100, where we handle local function scenario) and `CSharpCodeGenerationService.AddStatement` (line 467) is where the problem seems to be, But the logic there is complex enough that I couldn't extract/replicate a simple condition for when it won't work.